core/shared/src/main/scala/zio/internal/FiberRuntime.scala

```diff
@@ -1049,13 +1053,13 @@ final class FiberRuntime[E, A](fiberId: FiberId.Runtime, fiberRefs0: FiberRefs,
               // If we are nested inside another recursive call to `runLoop`,
               // then we need pop out to the very top in order to update
               // runtime flags globally:
-              cur = if (currentDepth > 0) {
+              if (currentDepth > 0) {
                 self.reifiedStack.ensureCapacity(currentDepth)
                 throw Trampoline(ZIO.unit, false)
               } else {
                 // We are at the top level, no need to update runtime flags
                 // globally:
-                ZIO.unit
+                cur = ZIO.unit
```

Comment:
It looks like you could get rid of the `else` clause and just assign to `cur`, since it is throwing on line 1058?

Reply:
so true